corehq/apps/hqmedia/view_helpers.py

```diff
@@ -76,16 +76,16 @@ def update_multimedia_paths(app, paths):
     for old_path, new_path in paths.items():
         for module in app.modules:
             success_counts[module.unique_id] += module.rename_media(old_path, new_path)
-            for form in module.forms:
+            for form in module.get_forms():
```

Comment:
This change (and the identical one below) are unrelated to this PR, it's just best practice to use `get_forms` so that the form's indexes are set correctly (see [IndexedSchema](https://github.com/dimagi/commcare-hq/blob/0753105b02359f4432b3a5b4eec9db1326a2483f/corehq/apps/app_manager/models.py#L249)).

Reply:
I don't understand the types of forms well enough but if we could do something like,
`def get_non_shadow_forms()`
or
`def get_forms_to_validate_multimedia_paths()`
on
`ModuleBase`
that would be nice.

Reply:
I'm going to hold off on adding a function for now, as this isn't an operation we do often.

I'd like to do a functionality-based test instead of checking form_type (like, we often use `hasattr` to check if a feature is available on a module, rather than checking the various module types) but haven't been able to come up with one, since shadow forms do have an xform, they just can't write to it.

Added f1eacd2d2cc to get the form_type check out of this file and into a form-specific class where it feels more appropriate.

Reply:
> I don't understand the types of forms well enough
like i said. Was just saying something which would be nice, if possible :)